### ISSUE TITLE
program hung when trying to use the result of ConditionalAccumulator's take_grad method after The dequeue operator

### ISSUE THREAD

**Issue by yaochengji** [tensorflow/tensorflow#12386]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes.
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 14.04 & Windows 10
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.2.1
- **Python version**: 3.5
- **Bazel version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

### Describe the problem
I tried to extend `SyncReplicasOptimizer` where I need to use the result of `ConditionalAccumulator`'s `take_grad` method after `sync_token_queue.dequeue()`, the program blocked. Then I looked into the code and thought that it might be caused  by  the step state in `ConditionalAccumulator`, but it still blocked.

### Source code / logs
I could modify the official example a bit to reproduce the problem.
The cmds I used are as below：
`python mnist_replica.py --job_name ps --task_index 0 --sync_replicas True`
`python mnist_replica.py --job_name worker --task_index 0 --sync_replicas True`
`python mnist_replica.py --job_name worker --task_index 1 --sync_replicas True`

Where `mnist_replica.py` only changed a line of importing module, [link](https://gist.github.com/yaochengji/455b541988920134df4260c954ad4985#file-mnist_replica-py). 
And I edited `sync_replicas_optimizer.py` a bit, two versions in [link1](https://gist.github.com/yaochengji/455b541988920134df4260c954ad4985#file-my_sync_replicas_optimizer_0-py) and in [link2](https://gist.github.com/yaochengji/455b541988920134df4260c954ad4985#file-my_sync_replicas_optimizer_1-py).

**Comment by drpngx** [comment]
We're trying to move to the `dataset` API which doesn't have these freezing issues. Typically it's either some racing condition or the eval queue not being started.

@nealwu @mrry @tfboyd do we have plans to convert our examples with the new API? I think that would remove a lot of confusion.

**Comment by mrry** [comment]
@drpngx I think the freezing issue is probably in the `tf.train.SyncReplicasOptimizer` rather than the queue-based input pipeline.

@josh11b and @isaprykin are working on a replacement that shouldn't have the same problems.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assigneee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @josh11b: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @josh11b: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @josh11b: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @josh11b: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @josh11b: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @josh11b: It has been 170 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @josh11b: It has been 185 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @josh11b: It has been 200 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @josh11b: It has been 215 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @josh11b: It has been 230 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @josh11b: It has been 245 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by josh11b** [comment]
Sorry for the lack of updates, we do not at this time have a direct replacement for SyncReplicaOptimizer. On one machine you can use tf.contrib.distribute.MirroredStrategy to do sync replication across multiple GPUs. Further sync replication options will be coming, but I'm not certain of the time frame.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @josh11b: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @josh11b: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @josh11b: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by harshini-gadige** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
